src/Microsoft.Azure.WebJobs.Host/Triggers/TriggerArgumentBinding/SimpleTriggerArgumentBinding.cs

```diff
@@ -82,5 +83,36 @@ public virtual async Task<ITriggerData> BindAsync(TTriggerValue value, ValueBind
 
             return triggerData;
         }
+
+        protected void InitializeBindingDataProvider()
```

Comment:
This type already has BindingDataContract/Hooks members that are used for contracts. You're adding another IBindingDataProvider + methods on the side, which will confuse things. Which should I use in a derived class?

What's the diference between CustomTriggerArgumentBinding and PocoTriggerArgumentBinding?

Reply:
`CustomTriggerArgumentBinding` is used if there are predefined convertors (Message -> string, Message -> byte[], etc.)
`PocoTriggerArgumentBinding` is used for PocoTypes.

It looks like changing base class `SimpleTriggerArgumentBinding` is a bad idea. I will update the PR with Hooks extension.

Reply:
Your initial PR was close I think. Anyhow, I've sent https://github.com/Azure/azure-webjobs-sdk/pull/2423 with some additional refactoring/cleanup